### PR TITLE
Inu 264 17 dress video camera

### PR DESCRIPTION
**Pull request by SanakaBarabaka**

### CODE REVIEW

File: VideoCamera/Source/SimulatorCamera.swift [hunk 1]
```diff
@@ -0,0 +1,239 @@
...
+                frame: CGRect(x: 0, y: 0, width: 100.0, height: 100.0),
+                flashColor: configuration.flashColor,
+                flashAnimationDuration: configuration.flashAnimationDuration,
+                image: simulatedImage)
+            videoFeedbackQueue = DispatchQueue(label: "com.shakuro.simulatorcamera.videofeedbackqueue")
+            videoFeedbackTimer = nil
+            videoFeedbackPixelBuffer = nil
+            videoFeedbackHandler = configuration.simulatedVideoFeedDelegate
+        } else {
+            throw VideoCameraError.invalidConfiguration(message: "simulated image is not provided")
```
Comment (vonipchenko) [inline VideoCamera/Source/SimulatorCamera.swift:50]: Do not trhow errror, please just provide default image

**Review by vonipchenko — CHANGES_REQUESTED** [review]

File: VideoCamera/Source/SimulatorCamera.swift (hunk 1 above)
Comment (SanakaBarabaka) [reply VideoCamera/Source/SimulatorCamera.swift:50]: moved to TODO to private repo

File: README.md [hunk 2]
```diff
@@ -1,2 +1,80 @@
...
+![Shakuro iOS Toolbox](title_image.svg)
+<br><br>
+
+![Version](https://img.shields.io/badge/version-0.5.4-blue.svg)
+![Platform](https://img.shields.io/badge/platform-iOS-lightgrey.svg)
+![License MIT](https://img.shields.io/badge/license-MIT-green.svg)
+
+Toolbox contains various components written in Swift.
+
+- [Components List](#components-list)
```
Comment (moseskimjj) [inline README.md]: Component List

File: README.md [hunk 3]
```diff
@@ -1,2 +1,80 @@
...
+- [Requirements](#requirements)
+- [Installation](#installation)
+- [License](#license)
+
+## Components List
```
Comment (moseskimjj) [inline README.md]: Component List

File: README.md [hunk 4]
```diff
@@ -1,2 +1,80 @@
...
+
+- [Device](/Device/)
+    - **DeviceType** - Helper for detecting model of the device / simulator.
```
Comment (moseskimjj) [inline README.md]: A helper for detecting the model of the device/simulator.

File: README.md [hunk 5]
```diff
@@ -1,2 +1,80 @@
...
+    - **DeviceOrientationListener** - Alternative for UIDevice.current.orientation.
```
Comment (moseskimjj) [inline README.md]: The alternative for UIDevice.current.orientation.

File: README.md [hunk 6]
```diff
@@ -1,2 +1,80 @@
...
+- [Extensions](/Extensions/) - Various extensions with small helper functions.
+- [ImageProcessing](/ImageProcessing/)
+    - **ImageProcessor** - Helper for CGImage and UIImage
```
Comment (moseskimjj) [inline README.md]: A helper for CGImage and UIImage.

File: README.md [hunk 7]
```diff
@@ -1,2 +1,80 @@
...
+- [Keychain](/Keychain/)
+    - **KeychainWrapper** - easy add/remove/get Codable object to/from Keychain.
```
Comment (moseskimjj) [inline README.md]: A wrapper to easily add, remove, or get a Codable object to/from Keychain.

File: README.md [hunk 8]
```diff
@@ -1,2 +1,80 @@
...
+- [Keyboard](/Keyboard/)
+    - **KeyboardHandler** - wrapper around keyboard notifications
```
Comment (moseskimjj) [inline README.md]: A wrapper around the keyboard notifications.

File: README.md [hunk 9]
```diff
@@ -1,2 +1,80 @@
...
+- [PlaceholderTextView](/PlaceholderTextView/) - UITextView subclass with a placeholder feature and ability to change own size depending on text contents
```
Comment (moseskimjj) [inline README.md]: A UITextView subclass with a placeholder feature and the ability to change own size depending on the text contents.

File: README.md [hunk 10]
```diff
@@ -1,2 +1,80 @@
...
+- [VideoCamera](/VideoCamera/) - wrapper for AVFoundation's camera.
```
Comment (moseskimjj) [inline README.md]: A wrapper for the AVFoundation camera.

File: README.md [hunk 11]
```diff
@@ -1,2 +1,80 @@
...
+
+- iOS 10.0+
+- Xcode 9.2+
+- Swift 4.0+
+
+## Installation
+
+### CocaPods
+
+[CocaPods](http://cocapods.org) is a dependency manager for Coca projects. You can install it with the following command:
```
Comment (moseskimjj) [inline README.md]: [CocoaPods](http://cocoapods.org) is a dependency manager for the Cocoa projects.

File: README.md [hunk 12]
```diff
@@ -1,2 +1,80 @@
...
+
+```bash
+$ gem install cocoapods
+```
+
+To integrate Toolbox into you Xcode project, specify it in your `Podfile`:
```
Comment (moseskimjj) [inline README.md]: To integrate Toolbox into your Xcode project, specify it in your `Podfile`:

File: README.md [hunk 13]
```diff
@@ -1,2 +1,80 @@
...
+target '<Your Target Name>' do
+    pod 'Shakuro.iOS_Toolbox/<Component Name>', :git => 'https://github.com/shakurocom/iOS_Toolbox', :tag => '0.5.4'
+#example:
+    pod 'Shakuro.iOS_Toolbox/Keychain', :git => 'https://github.com/shakurocom/iOS_Toolbox', :tag => '0.5.4'
+end
+```
+
+### Manually
+
+If you prefer to not use CocoPods, than you can integrate any/all components from Shakuro iOS Toolbox simply by copying them to your project
```
Comment (moseskimjj) [inline README.md]: If you prefer not to use CocoaPods, you can integrate any/all components from the Shakuro iOS Toolbox simply by copying them to your project.

File: README.md [hunk 14]
```diff
@@ -1,2 +1,80 @@
...
+end
+```
+
+Then, run the following command:
+
+```bash
+$ pod install
+```
+
+You can use integrate only needed components. To do this you need to specify subpod:
```
Comment (moseskimjj) [inline README.md]: You can use/integrate only the necessary components. To do this, you need to specify the subpod:

**Pull request merged**